### DISCUSSION TITLE
How to check out a PR to a local branch?

### DISCUSSION THREAD

**Discussion by larspalo** [GrandOrgue/grandorgue#1383] [Developers]
@oleg68 @rousseldenis Sometimes, especially when the automatic builds fail, I'd like to test a PR locally in a new separate branch (to avoid trouble). Anyone that have an idea how I could do that? Feel free to educate me! For the code button on the PR, I see a github cli command, but I'm not using github cli - rather the normal git.

**Comment by oleg68** [comment]
You have to add my fork repo as a new named remote to your local git repo. After that you can do everything with fetching, switching branchs, checking out except pushing. You can ever make a PR from your current fork to a branch in my fork. See 'git remote --help' for hints.

**Reply by oleg68** [reply]
More details:

Open a terminal and go to your local GrandOrgue repo

```
git remote add oleg68 https://github.com/oleg68/GrandOrgue-official.git
git fetch oleg68
git checkout --track oleg68/feature/new-organ-settings
```

**Reply by larspalo** [reply]
Since I have the official grandorgue repo set as upstream would it work to do something like:

git fetch upstream pull/1382/head:Test1382

and then just switch to the Test1382 branch? What would be the advantages of adding your fork as another remote compared to just fetching the pr to a separate branch in my own local git repo?

**Reply by larspalo** [reply]
@oleg68 Thanks for the help. It seems to have worked with the git fetch command I cited above, but since I'm still very much learning about git I cant help but wonder if it's somehow better to do it as you suggested and add your fork as a new remote?

**Reply by oleg68** [reply]
> git fetch upstream pull/1382/head:Test1382

I haven't known that it is possible. I thought that my branch did not exist in the upstream repository (because ``git branch -r`` did not show it) and it existed only in my fork.

**Reply by larspalo** [reply]
I had to google it! :-) But since I'm still a newbie when it comes to understanding git I wanted to ask before doing. I think it worked because I had set the upstream to the official grandorgue repo earlier.
